cmd/flow-dps-live/main.go

```diff
@@ -0,0 +1,401 @@
+	level, err := zerolog.ParseLevel(flagLevel)
+	if err != nil {
+		log.Error().Str("level", flagLevel).Err(err).Msg("could not parse log level")
+		return failure
+	}
+	log = log.Level(level)
+
+	// Ensure that at least one index is specified.
+	if !flagIndexAll && !flagIndexCommit && !flagIndexHeader && !flagIndexPayloads && !flagIndexCollections &&
+		!flagIndexGuarantees && !flagIndexTransactions && !flagIndexResults && !flagIndexEvents && !flagIndexSeals {
+		log.Error().Str("level", flagLevel).Msg("no indexing option specified, use -a/--all to build all indexes")
+		pflag.Usage()
+		return failure
+	}
+
+	// Fail if IndexAll is specified along with other index flags, as this would most likely mean that the user does
+	// not understand what they are doing.
+	if flagIndexAll && (flagIndexCommit || flagIndexHeader || flagIndexPayloads || flagIndexGuarantees ||
+		flagIndexCollections || flagIndexTransactions || flagIndexResults || flagIndexEvents || flagIndexSeals) {
+		log.Error().Str("level", flagLevel).Msg("-a/--all is mutually exclusive with specific indexing flags")
+		pflag.Usage()
+		return failure
+	}
+
+	// Open index database.
+	db, err := badger.Open(dps.DefaultOptions(flagIndex))
+	if err != nil {
+		log.Error().Str("index", flagIndex).Err(err).Msg("could not open index DB")
+		return failure
+	}
+	defer db.Close()
+
+	// Open protocol state database.
+	data, err := badger.Open(dps.DefaultOptions(flagData))
+	if err != nil {
+		log.Error().Err(err).Msg("could not open blockchain database")
+		return failure
+	}
+	defer data.Close()
+
+	// We initialize a metrics logger regardless of whether metrics are enabled;
+	// it will just do nothing if there are no registered metrics.
+	mout := output.New(log, flagMetricsInterval)
+
+	// The storage library is initialized with a codec and provides functions to
+	// interact with a Badger database while encoding and compressing
+	// transparently.
+	var codec dps.Codec
+	codec = zbor.NewCodec()
+	if flagMetrics {
+		size := rcrowley.NewSize("store")
+		mout.Register(size)
+		codec = metrics.NewCodec(codec, size)
+	}
+	storage := storage.New(codec)
+
+	// Check if index already exists.
+	_, err = index.NewReader(db, storage).First()
+	indexExists := err == nil
+	if indexExists && !flagForce {
+		log.Error().Err(err).Msg("index already exists, manually delete it or use (-f, --force) to overwrite it")
+		return failure
+	}
+
+	// The loader component is responsible for loading and decoding the checkpoint.
+	load := loader.New(
+		loader.WithCheckpointPath(flagCheckpoint),
+	)
+
+	nodeID, err := flow.HexStringToIdentifier(flagNodeID)
+	if err != nil {
+		log.Error().Err(err).Msg("invalid node ID")
+		return failure
+	}
+
+	ctx := context.Background()
+	client, err := googlecloud.NewClient(ctx)
+	if err != nil {
+		log.Error().Err(err).Msg("could not connect to Google Cloud Platform")
+		return failure
+	}
+
+	bootstrapIdentities, err := network.RetrieveIdentities(flagBootstrap)
```

Comment:
That's exactly what there was before, and then from what we said, I understood that we needed to give all of the bootstrap identities to the consensus follower, either by pulling them from GCP (which we talked about, and then I was told was not necessary, and that just using the file would be enough) or by parsing the file directly from the filesystem.

It's not the first time that there's such a big misunderstanding between us like this, I think we really need to work on it because on this PR alone we lost a lot of time with me doing useless stuff and reverting it over and over 😅 Even when I take notes on what we say, it turns out that what I wrote is most of the time not what you meant, or even the polar opposite.